### DISCUSSION TITLE
Why is an element of a symbolic vector is not reported  as true by is_symbolic()?

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3479] [Q&A] [answered]
```
m=cs.MX.sym('m',2,1)
print('sym vector',m.is_symbolic())
print('element of sym vector',m[0].is_symbolic())
```

**Comment by jgillis** [comment, marked as answer]
`is_symbolic `check if an expression is a pure symbol.
Pure means that no operations should happen on the symbol
(not even vec, transpose, index, concatenation, ...)
By consequence, a slice of a vector-shaped MX symbol is not a symbol.
However, the SX type is really a container format with scalar entries.
Therefore, a slice of a vector-shaped SX symbol is still a symbol.

Related is `is_valid_input`. It is more forgiving than is_symbolic.
Some compositions are allowed: vec, vertcat.
